console-subscriber/src/aggregator.rs

```diff
@@ -291,28 +476,52 @@ impl Aggregator {
         } else {
             None
         };
-        let new_tasks = self
-            .tasks
-            .since_last_update()
-            .map(|(id, task)| task.to_proto(id.clone()))
-            .collect();
-        let now = SystemTime::now();
-        let stats_update = self
-            .stats
-            .since_last_update()
-            .map(|(id, stats)| (id.into_u64(), stats.to_proto()))
-            .collect();
 
-        let update = proto::tasks::TaskUpdate {
-            new_metadata,
-            new_tasks,
-            stats_update,
+        let now = SystemTime::now();
+        let update = proto::instrument::InstrumentUpdate {
             now: Some(now.into()),
+            new_metadata,
+            task_update: Some(proto::tasks::TaskUpdate {
+                new_tasks: self
+                    .tasks
+                    .as_proto(Include::UpdatedOnly)
+                    .values()
+                    .cloned()
+                    .collect(),
+                stats_update: self.task_stats.as_proto(Include::UpdatedOnly),
+            }),
+            resource_update: Some(proto::resources::ResourceUpdate {
+                new_resources: self
+                    .resources
+                    .as_proto(Include::UpdatedOnly)
+                    .values()
+                    .cloned()
+                    .collect(),
+                stats_update: self.resource_stats.as_proto(Include::UpdatedOnly),
+            }),
+            async_op_update: Some(proto::async_ops::AsyncOpUpdate {
+                new_async_ops: self
+                    .async_ops
+                    .as_proto(Include::UpdatedOnly)
+                    .values()
+                    .cloned()
+                    .collect(),
+                stats_update: self.async_op_stats.as_proto(Include::UpdatedOnly),
+            }),
+            resource_op_update: Some(proto::resource_ops::ResourceOpUpdate {
+                new_resource_ops: self
+                    .resource_ops
+                    .as_proto(Include::UpdatedOnly)
+                    .values()
+                    .cloned()
+                    .collect(),
```

Comment:
similarly to my above comment, we can eliminate an unnecessary `collect` and `clone` call if we don't use `IdData::as_proto` when we don't want the values:
```suggestion
                new_tasks: self
                    .tasks
                    .since_last_update()
                    .map(|(_, value)| value.to_proto())
                    .collect(),
                stats_update: self.task_stats.as_proto(Include::UpdatedOnly),
            }),
            resource_update: Some(proto::resources::ResourceUpdate {
                new_resources: self
                    .resources
                    .since_last_update()
                    .map(|(_, value)| value.to_proto())
                    .collect(),
                stats_update: self.resource_stats.as_proto(Include::UpdatedOnly),
            }),
            async_op_update: Some(proto::async_ops::AsyncOpUpdate {
                new_async_ops: self
                    .async_ops
                    .since_last_update()
                    .map(|(_, value)| value.to_proto())
                    .collect(),
                stats_update: self.async_op_stats.as_proto(Include::UpdatedOnly),
            }),
            resource_op_update: Some(proto::resource_ops::ResourceOpUpdate {
                new_resource_ops: self
                    .resource_ops
                    .since_last_update()
                    .map(|(_, value)| value.to_proto())
                    .collect(),
```